scene/3d/skeleton_3d.cpp

```diff
@@ -453,102 +517,121 @@ int Skeleton3D::get_bone_count() const {
-	ERR_FAIL_INDEX(p_bone, bones.size());
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX(p_bone, bone_size);
 
 	_update_process_order();
 
 	int parent = bones[p_bone].parent;
 	while (parent >= 0) {
-		bones.write[p_bone].rest = bones[parent].rest * bones[p_bone].rest;
+		bones[p_bone].rest = bones[parent].rest * bones[p_bone].rest;
 		parent = bones[parent].parent;
 	}
 
-	bones.write[p_bone].parent = -1;
+	bones[p_bone].parent = -1;
 	process_order_dirty = true;
 
 	_make_dirty();
 }
 
 void Skeleton3D::set_bone_disable_rest(int p_bone, bool p_disable) {
-	ERR_FAIL_INDEX(p_bone, bones.size());
-	bones.write[p_bone].disable_rest = p_disable;
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX(p_bone, bone_size);
+	bones[p_bone].disable_rest = p_disable;
 }
 
 bool Skeleton3D::is_bone_rest_disabled(int p_bone) const {
-	ERR_FAIL_INDEX_V(p_bone, bones.size(), false);
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX_V(p_bone, bone_size, false);
 	return bones[p_bone].disable_rest;
 }
 
 int Skeleton3D::get_bone_parent(int p_bone) const {
-	ERR_FAIL_INDEX_V(p_bone, bones.size(), -1);
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX_V(p_bone, bone_size, -1);
 
 	return bones[p_bone].parent;
 }
 
-void Skeleton3D::set_bone_rest(int p_bone, const Transform3D &p_rest) {
-	ERR_FAIL_INDEX(p_bone, bones.size());
+Vector<int> Skeleton3D::get_bone_children(int p_bone) const {
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX_V(p_bone, bone_size, Vector<int>());
+	return bones[p_bone].child_bones;
+}
 
-	bones.write[p_bone].rest = p_rest;
+void Skeleton3D::set_bone_children(int p_bone, Vector<int> p_children) {
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX(p_bone, bone_size);
+	bones[p_bone].child_bones = p_children;
+
+	process_order_dirty = true;
 	_make_dirty();
 }
-Transform3D Skeleton3D::get_bone_rest(int p_bone) const {
-	ERR_FAIL_INDEX_V(p_bone, bones.size(), Transform3D());
 
-	return bones[p_bone].rest;
+void Skeleton3D::add_bone_child(int p_bone, int p_child) {
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX(p_bone, bone_size);
+	bones[p_bone].child_bones.push_back(p_child);
+
+	process_order_dirty = true;
+	_make_dirty();
 }
 
-void Skeleton3D::set_bone_enabled(int p_bone, bool p_enabled) {
-	ERR_FAIL_INDEX(p_bone, bones.size());
+void Skeleton3D::remove_bone_child(int p_bone, int p_child) {
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX(p_bone, bone_size);
 
-	bones.write[p_bone].enabled = p_enabled;
+	int child_idx = bones[p_bone].child_bones.find(p_child);
+	if (child_idx >= 0) {
+		bones[p_bone].child_bones.remove(child_idx);
+	} else {
+		WARN_PRINT("Cannot remove child bone: Child bone not found.");
+	}
+
+	process_order_dirty = true;
 	_make_dirty();
 }
 
-bool Skeleton3D::is_bone_enabled(int p_bone) const {
-	ERR_FAIL_INDEX_V(p_bone, bones.size(), false);
-	return bones[p_bone].enabled;
+Vector<int> Skeleton3D::get_parentless_bones() const {
```

Comment:
Same as above, this should return `const LocalVector &`, since `parentless_bones` is now `LocalVector`, if you can't do the change, please make sure to make `parentless_bones` a `Vector` again.

Depending how you solve it, make sure to adjust its usage too.